### DISCUSSION TITLE
Extract classes common across component variants

### DISCUSSION THREAD

**Discussion by dhnaranjo** [ruby-ui/ruby_ui#47] [Ideas]
I'd like it if the set of classes common across the variants of a component were pulled into something like a `common_classes` method. Take `PhlexUI::Link`, where each of the variants list of classes includes `whitespace-nowrap, inline-flex, items-center, justify-center, rounded-md, font-medium, transition-colors, focus-visible:outline-none, focus-visible:ring-1, focus-visible:ring-ring, disabled:pointer-events-none, disabled:opacity-50`. If these were broke up I could better see what the actual differences are.

**Comment by dhnaranjo** [comment]
I wonder how this could actually be maintained other than just like... as a policy as you write components. Some kinda custom cop? That seems like. lot. Maybe this isn't an issue but a like, discussion topic?

**Comment by sethhorsley** [comment]
Hey @dhnaranjo this is a good idea.
At first i think the idea was to keep things as close to how shadcn has elements structured.

I am not against this and we should explore how we could do this while keeping maintaince in mind.
